### PR TITLE
Add ResultStoreConfig to ProwJobDefault

### PR DESCRIPTION
**Pull request by airbornepony**
Improve configuration efficiency and flexibility by configuring ResultStore uploading in the ProwJobDefault configs.

Avoid "Reporter" language since the current Crier reporter is an implementation detail that may change in the future, e.g. if we decide to implement streaming uploads.

Complete test coverage in ProwJobDefault.ApplyDefault.

The current config is retained until Alpha configurations are migrated to use ProwJobDefault.

### CODE REVIEW

File: prow/plugins/plugin-config-documented.yaml [hunk 1]
```diff
@@ -294,6 +294,18 @@ bugzilla:
 cat:
     # Path to file containing an api key for thecatapi.com
     key_path: ' '
+cherry_pick_approved:
```
[the remark this answers is not in the document]
Comment (airbornepony) [reply prow/plugins/plugin-config-documented.yaml]: Removed from this PR.

File: prow/apis/prowjobs/v1/types.go [hunk 2]
```diff
@@ -354,8 +354,8 @@ type ReporterConfig struct {
 	Slack       *SlackReporterConfig `json:"slack,omitempty"`
 }
 
-// TODO: This config is used only for alpha testing and will
-// likely move to ProwJobDefaults for flexibility.
+// TODO: This config was used for alpha testing and is now replaced
```
[the remark this answers is not in the document]
Comment (airbornepony) [reply prow/apis/prowjobs/v1/types.go:357]: Added a mention of this. Hope to have it gone very soon.

**Comment by listx** [comment]
/lgtm
/hold

Unhold when you want to. Thanks.

**Pull request merged**